auto-complete.js

```diff
@@ -117,50 +134,67 @@ var autoComplete = (function(){
             }
             that.parentElement.appendChild(that.sc);
 
-            live('autocomplete-suggestion', 'mouseleave', function(e){
+            live('autocomplete-suggestion', 'mouseleave', function (e) {
                 var sel = that.sc.querySelector('.autocomplete-suggestion.selected');
-                if (sel) setTimeout(function(){ sel.className = sel.className.replace('selected', ''); }, 20);
+                if (sel) setTimeout(function () { sel.className = sel.className.replace('selected', ''); }, 20);
             }, that.sc);
 
-            live('autocomplete-suggestion', 'mouseover', function(e){
+            live('autocomplete-suggestion', 'mouseover', function (e) {
                 var sel = that.sc.querySelector('.autocomplete-suggestion.selected');
                 if (sel) sel.className = sel.className.replace('selected', '');
                 this.className += ' selected';
             }, that.sc);
 
-            live('autocomplete-suggestion', 'mousedown', function(e){
-                if (hasClass(this, 'autocomplete-suggestion')) { // else outside click
-                    var v = this.getAttribute('data-val');
+            live('autocomplete-suggestion-text', 'mousedown', function (e) {
+                if (hasClass(this, 'autocomplete-suggestion-text')) { // else outside click
+                    var v = this.parentElement.getAttribute('data-val');
+                    var index = this.parentElement.getAttribute('data-index');
+                    if (o.queryHistoryStorageName) {
+                        addQueryToLocalStorage(o.queryHistoryStorageName, rawData[index]);
+                    }
                     that.value = v;
                     o.onSelect(e, v, this);
                     that.sc.style.display = 'none';
                 }
             }, that.sc);
 
-            that.blurHandler = function(){
-                try { var over_sb = document.querySelector('.autocomplete-suggestions:hover'); } catch(e){ var over_sb = 0; }
+            live('autocomplete-suggestion--local-remove-button', 'mousedown', function (e) {
+                var index = this.parentElement.getAttribute('data-index');
+                this.parentElement.remove()
```

Comment:
Not supported by any IE https://caniuse.com/#feat=childnode-remove

Use [removeChild](https://developer.mozilla.org/en-US/docs/Web/API/Node/removeChild)
```suggestion
                this.parentElement.parentElement.removeChild(this.parentElement);
```